src/Aspire.Dashboard/Otlp/Storage/TelemetryRepository.cs

```diff
@@ -14,25 +14,40 @@
 
 namespace Aspire.Dashboard.Otlp.Storage;
 
-public class TelemetryRepository
+public sealed class TelemetryOptions
 {
-    internal const string MaxLogCountKey = "MaxLogCount";
-    internal const string MaxTraceCountKey = "MaxTraceCount";
-    internal const string MaxMetricsCountKey = "MaxMetricsCount";
+    public int LogCountLimit { get; init; }
+    public int TraceCountLimit { get; init; }
+    public int MetricsCountLimit { get; init; }
+    public int AttributeCountLimit { get; init; }
+    public int AttributeLengthLimit { get; init; }
+    public int SpanEventCountLimit { get; init; }
+}
 
-    private const int DefaultMaxTelemetryCount = 10_000;
-    private const int DefaultMaxMetricsCount = 100_000; // Allows for 1 metric point per second for over 24 hours.
+public sealed class TelemetryRepository
+{
+    internal const string LogCountLimitKey = "DOTNET_DASHBOARD_OTEL_LOG_COUNT_LIMIT";
+    internal const string TraceCountLimitKey = "DOTNET_DASHBOARD_OTEL_TRACE_COUNT_LIMIT";
+    internal const string MetricsCountLimitKey = "DOTNET_DASHBOARD_OTEL_METRIC_COUNT_LIMIT";
+    internal const string AttributeCountLimitKey = "DOTNET_DASHBOARD_OTEL_ATTRIBUTE_COUNT_LIMIT";
+    internal const string AttributeLengthLimitKey = "DOTNET_DASHBOARD_OTEL_ATTRIBUTE_LENGTH_LIMIT";
+    internal const string SpanEventCountLimitKey = "DOTNET_DASHBOARD_OTEL_SPAN_EVENT_COUNT_LIMIT";
+
+    private const int DefaultTraceCountLimit = 10_000;
+    private const int DefaultMetricsCountLimit = 50_000; // Allows for 1 metric point per second for over 12 hours.
```

Comment:
Do we know the dashboard is still using a reasonable amount of memory when these limits are both reached (max items all max length)?

Reply:
No. I'm sure that someone who really wants to max memory could use the worst-case scenario (10,000 log messages, each with 128 attributes, each with a large value) to use a lot of memory.

However, the limits are all configurable, so a hoster has the option to place smaller limits.

The limits will help prevent one errant log message (megabyte sized message) from using excessive memory and keep the dashboard usable.